vars/unitTestPost.groovy

```diff
@@ -45,45 +52,51 @@ def call(Map config = [:]) {
     return
   }
 
-  junit testResults: config.get('testResults', 'test_results/*.xml')
+  def test_results = config.get('testResults', 'test_results/*.xml')
+  if (test_results != "") {
+    junit testResults: test_results
+  }
 
   def artifact_list = config.get('artifacts', ['run_test.sh/*', 'vm_test/**'])
   artifact_list.each {
     archiveArtifacts artifacts: it
   }
-
+  
+  def valgrind_pattern = config.get('valgrind_pattern', 'dnt.*.memcheck.xml')
   publishValgrind failBuildOnInvalidReports: true,
                   failBuildOnMissingReports: true,
                   failThresholdDefinitelyLost: '0',
                   failThresholdInvalidReadWrite: '0',
                   failThresholdTotal: '0',
-                  pattern: 'dnt.*.memcheck.xml',
+                  pattern: valgrind_pattern,
                   publishResultsForAbortedBuilds: false,
                   publishResultsForFailedBuilds: true,
                   sourceSubstitutionPaths: '',
                   unstableThresholdDefinitelyLost: '0',
                   unstableThresholdInvalidReadWrite: '0',
                   unstableThresholdTotal: '0'
-
-  recordIssues enabledForFailure: true,
-               failOnError: true,
-               referenceJobName: config.get('referenceJobName',
-                                            'daos-stack/daos/master'),
-               ignoreFailedBuilds: false,
-               ignoreQualityGate: true,
-               // Set qualitygate to 1 new "NORMAL" priority message
-               // Supporting messages to help identify causes of
-               // problems are set to "LOW", and there are a
-               // number of intermittent issues during server
-               // shutdown that would normally be NORMAL but in
-               // order to have stable results are set to LOW.
-
-               qualityGates: [
-                 [threshold: 1, type: 'TOTAL_HIGH', unstable: true],
-                 [threshold: 1, type: 'TOTAL_ERROR', unstable: true],
-                 [threshold: 1, type: 'NEW_NORMAL', unstable: true]],
-                name: "Node local testing",
-                tool: issues(pattern: 'vm_test/nlt-errors.json',
-                             name: 'NLT results',
-                             id: 'VM_test')
+  if (valgrind == '') {
```

Comment:
if (stage_info['valgrind']) {

This tests false when the 'valgrind' member does not exist, or is set to a value that evaluates as false such as an empty string.

Reply:
right, no, this should execute when valgrind is 'disabled', I'll fix it next patch